diff -iwBEr -U 4 busybox-master/arch/i386/Makefile busybox-master-20131016/arch/i386/Makefile
--- busybox-master/arch/i386/Makefile	2013-10-16 12:43:30.000000000 +0000
+++ busybox-master-20131016/arch/i386/Makefile	2013-11-10 17:34:25.000000000 +0000
@@ -3,5 +3,6 @@
 # ==========================================================================
 
 # -mpreferred-stack-boundary=2 is essential in preventing gcc 4.2.x
 # from aligning stack to 16 bytes. (Which is gcc's way of supporting SSE).
-CFLAGS += $(call cc-option,-march=i386 -mpreferred-stack-boundary=2,)
+# CFLAGS += $(call cc-option,-march=i386 -mpreferred-stack-boundary=2,)
+CFLAGS += $(call cc-option,$(CONFIG_EXTRA_CFLAGS))
diff -iwBEr -U 4 busybox-master/util-linux/volume_id/volume_id_internal.h busybox-master-20131016/util-linux/volume_id/volume_id_internal.h
--- busybox-master/util-linux/volume_id/volume_id_internal.h	2013-10-16 12:43:30.000000000 +0000
+++ busybox-master-20131016/util-linux/volume_id/volume_id_internal.h	2013-10-17 23:29:44.000000000 +0000
@@ -91,11 +91,13 @@
 
 /* util.h */
 
 /* size of superblock buffer, reiserfs block is at 64k */
-#define SB_BUFFER_SIZE				0x11000
+/* HD drives >= 1 TB need for some reason doubled buffer sizes  */
+#define SB_BUFFER_SIZE				0x21000
 /* size of seek buffer, FAT cluster is 32k max */
-#define SEEK_BUFFER_SIZE			0x10000
+/* to appear with BLKID applet */
+#define SEEK_BUFFER_SIZE			0x20000
 
 #if BB_LITTLE_ENDIAN
 # define le16_to_cpu(x) (uint16_t)(x)
 # define le32_to_cpu(x) (uint32_t)(x)
